### PR TITLE
docs: replace double dashes with m-dashes, fix typos

### LINKED ISSUE #46173: Double dashes instead of m-dashes and typo
### Description

there are two places on [What is Angular? page](https://angular.io/guide/what-is-angular) where double dashes are used instead of m-dashes
Screenshot 2022-05-28 at 12 46 33 p m
Screenshot 2022-05-28 at 12 46 40 p m

there is an unnecessary space before a m-dash on [Complex animations sequences page](https://angular.io/guide/complex-animation-sequences)
Screenshot 2022-05-28 at 12 48 47 p m

there is a typo on [Route transition animations page](https://angular.io/guide/route-animations), the last dot shouldn’t be here
Screenshot 2022-05-28 at 12 50 59 p m

### What is the affected URL?

https://angular.io/guide/what-is-angular https://angular.io/guide/complex-animation-sequences https://angular.io/guide/route-animations

### Please provide the steps to reproduce the issue

go to [What is Angular? page](https://angular.io/guide/what-is-angular) and search for ‘--’

go to [Complex animations sequences page](https://angular.io/guide/complex-animation-sequences) and search for ‘—’

go to [Route transition animations page](https://angular.io/guide/route-animations) and search for ‘data?.’

### Please provide the expected behavior vs the actual behavior you encountered

m-dashes are used instead of double dashes and typos aren’t present 

### Please provide a screenshot if possible

_No response_

### Please provide the exception or error you saw

_No response_

### Is this a browser-specific issue? If so, please specify the device, browser, and version.

_No response_

### PR DESCRIPTION
**Pull request by looshch**
Fixes #46173

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #46173 

## What is the new behavior?

m-dashes are used instead of double dashes and typos aren’t present

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/examples/animations/src/app/app.component.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class AppComponent {
   constructor(private contexts: ChildrenOutletContexts) {}
 
   getRouteAnimationData() {
-    return this.contexts.getContext('primary')?.route?.snapshot?.data?.['animation'];
+    return this.contexts.getContext('primary')?.route?.snapshot?.data?['animation'];
```
Comment (gkalpak) [inline aio/content/examples/animations/src/app/app.component.ts:41]: This change is actually invalid. When accessing properties using square bracket notation, the [correct syntax][1] for optional chaining is indeed `o?.[prop]`.

Can you please revert this change :pray:

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#syntax

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @looshch!
I requested one change. Otherwise lgtm :+1:

**Review by gkalpak — APPROVED** [review]
🚀

**Review by gkalpak — APPROVED** [review]
Reviewed-for: global-docs-approvers

**Review by jessicajaniuk — APPROVED** [review]
reviewed for: fw-animations, docs-getting-started-and-tutorial

**Pull request closed without merging**